### PR TITLE
feat: flush writes to http response on every chunk

### LINKED ISSUE #742: Keda Http Add-on support streaming
### Discussed in https://github.com/kedacore/http-add-on/discussions/736

<div type='discussions-op-text'>

<sup>Originally posted by **snirbenyosef** July 21, 2023</sup>
Hey,
Thank you for this awesome addon!

i have a question:
before using the addon I used my pod (Creating virtual Service,etc) as Streaming response, i mean i send requests to the pod, and i get my answer as streaming.
after using the Addon and setting up the inspector of keda html addon, the streaming stopped and i'm getting the full response and not streaming in batches.

do Keda Http add-on support Streaming right now?

thanks.
</div>

### PR DESCRIPTION
**Pull request by itayariel**
This pr enables streaming responses using the proxy code
Currently the logging middleware breaks streaming and responses are sent only upon completion

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [ ] Any necessary documentation is added, such as:
  - [`README.md`](/README.md)
  - [The `docs/` directory](./docs)
  - [The docs repo](https://github.com/kedacore/keda-docs)

Fixes #742

### CODE REVIEW

**Review by JorTurFer — COMMENTED** [review]
Nice improvement! ❤️ 
Could you introduce an entry in changelog and any test to cover this scenario? If this is coverable with unit test it'd be nice, if not, we also execute e2e test on each PR/commit.

There is also a problem with DCO, could you sign your commits please? it's a requirement for us

**Comment by itayariel** [comment]
Thanks will work on your comments

File: CHANGELOG.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ This changelog keeps track of work items that have been completed and are ready
 - **Operator**: Remove ScaledObject `name` & `app` custom labels ([#717](https://github.com/kedacore/http-add-on/issues/717))
 - **Scaler**: Provide graceful shutdown for grpc server on SIGINT and SIGTERM ([#731](https://github.com/kedacore/http-add-on/issues/731))
 - **Scaler**: remplement custom interceptor metrics ([#718](https://github.com/kedacore/http-add-on/issues/718))
+- **Interceptor**: Add support for streaming responses ([#726](https://github.com/kedacore/http-add-on/issues/743))
```
Comment (JorTurFer) [inline CHANGELOG.md:37]: Please, sort this alphabetically and update the id too
```suggestion
- **Interceptor**: Add support for streaming responses ([#742](https://github.com/kedacore/http-add-on/issues/https://github.com/kedacore/http-add-on/issues/742))
```

**Review by JorTurFer — COMMENTED** [review]
Could you add a unit test for this?

File: CHANGELOG.md (hunk 1 above)
Comment (itayariel) [reply CHANGELOG.md]: fixed

**Comment by itayariel** [comment]
hey @JorTurFer I tried to write unit tests, but didn't a find nice a way to test the fix, do you have any suggestion?

File: CHANGELOG.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ This changelog keeps track of work items that have been completed and are ready
 - **Operator**: Remove ScaledObject `name` & `app` custom labels ([#717](https://github.com/kedacore/http-add-on/issues/717))
 - **Scaler**: Provide graceful shutdown for grpc server on SIGINT and SIGTERM ([#731](https://github.com/kedacore/http-add-on/issues/731))
 - **Scaler**: remplement custom interceptor metrics ([#718](https://github.com/kedacore/http-add-on/issues/718))
-- **Interceptor**: Add support for streaming responses ([#726](https://github.com/kedacore/http-add-on/issues/743))
+- **Interceptor**: Add support for streaming responses ([#743](https://github.com/kedacore/http-add-on/issues/742))
```
Comment (JorTurFer) [inline CHANGELOG.md:37]:
```suggestion
- **Interceptor**: Add support for streaming responses ([#743](https://github.com/kedacore/http-add-on/issues/743))
```

**Comment by JorTurFer** [comment]
> hey @JorTurFer I tried to write unit tests, but didn't a find nice a way to test the fix, do you have any suggestion?

The recorder used for the logginResponseWritter unit tests has a flag to know if flush has been called `Flushed`. I guess that thanks to this check in unit tests, and the e2e tests, we can have at least a minimum guarantee but if you could write an e2e test for this feature would be even better.

**Comment by JorTurFer** [comment]
any update on this?

**Review by JorTurFer — APPROVED** [review]

**Pull request merged**